guidance/d_star_lite/d_star_lite/d_star_lite.py

```diff
@@ -0,0 +1,408 @@
+        
+        compare_keys(key_pair1: tuple[float, float], key_pair2: tuple[float, float]) -> bool: Compares the priority keys of two nodes.
+        
+        compute_shortest_path(): Computes or recomputes the shortest path from the start to the goal using the D* Lite algorithm.
+        
+        compute_current_path() -> list[DSLNode]: Computes the current path from the start to the goal.
+        
+        get_WP() -> list[list[int]]: Retrieves the waypoints and adjusts their coordinates to the original coordinate system.
+        
+        dsl_main(start: DSLNode, goal: DSLNode) -> tuple[bool, list[int], list[int]]: Main function to run the D* Lite algorithm.
+    """
+
+    motions = [ # Represents the possible motions in the grid and corresponding costs
+        DSLNode(1, 0, 1), DSLNode(0, 1, 1), DSLNode(-1, 0, 1), DSLNode(0, -1, 1), 
+        DSLNode(1, 1, math.sqrt(2)), DSLNode(1, -1, math.sqrt(2)),
+        DSLNode(-1, 1, math.sqrt(2)), DSLNode(-1, -1, math.sqrt(2))
+    ]
+
+    def __init__(self, ox: list, oy: list, start: DSLNode, goal: DSLNode, dist_to_obstacle: float = 4.5, origin: tuple = (0,0), height: int = 25, width: int = 25):
+        """
+        Initializes a new instance of the DStarLite class.
+
+        Args:
+            ox (list): The x-coordinates of the obstacles.
+            oy (list): The y-coordinates of the obstacles.
+            start (DSLNode): The start node.
+            goal (DSLNode): The goal node.
+            dist_to_obstacle (float): The minimum distance a DSLNode must be from any obstacle to be considered valid. Defaults to 4.5.
+            origin (tuple): The origin of the grid. Defaults to (0, 0).
+            height (int): The height of the grid. Defaults to 25.
+            width (int): The width of the grid. Defaults to 25.
+        """
+        if len(ox) != len(oy):
+            raise ValueError("The number of x and y coordinates must be equal.")
+        
+        if len(ox) == 0 and len(oy) == 0: # If no obstacles are provided
+            self.obstacles = []
+        else:
+            self.obstacles = [DSLNode(x, y) for x, y in zip(ox, oy)] # The obstacles as nodes
+            self.obstacles_xy = np.array( # The obstacles as xy coordinates
+                [[obstacle.x, obstacle.y] for obstacle in self.obstacles]
+            )
+        
+        self.start = start # The start DSLNode
+        self.goal = goal # The goal DSLNode
+        self.origin = origin # The origin of the world grid
+        self.height = height # The height of the world grid
+        self.width = width # The width of the world grid
+
+        # Compute the min and max values for the world boundaries
+        self.x_min = int(origin[0]-self.width)
+        self.y_min = int(origin[1]-self.height)
+        self.x_max = int(origin[0]+self.width)
+        self.y_max = int(origin[1]+self.height)
```

Comment:
Remember to divide self.width and self.height by 2 if you want the requested width and height.